src/coreclr/src/inc/corinfoinstructionset.h

```diff
@@ -59,6 +59,7 @@ enum CORINFO_InstructionSet
     InstructionSet_SSE2_X64=25,
     InstructionSet_SSE41_X64=26,
     InstructionSet_SSE42_X64=27,
+    InstructionSet_AVX512=28,
```

Comment:
This needs to be `AVX512F` to match the actual ISA name. There are a number of AVX512 extensions and it will be important to differentiate them if we eventually extend HWIntrinsics to support them.

Reply:
Agreed, this is indeed the naming on my original repo:
MSVC:
https://github.com/damageboy/vxsort-cpp/blob/master/vxsort/isa_detection_msvc.cpp#L53

clang:
https://github.com/damageboy/vxsort-cpp/blob/master/vxsort/isa_detection_sane.cpp#L20

Reply:
Would it be better for GC to have its own local detection (e.g. based on @damageboy original repo), and not could it with the JIT/EE interface and intrinsics support in the JIT?

This instruction set enum is autogenerated (look for "DO NOT EDIT THIS FILE! IT IS AUTOGENERATE" at the top of the file). Adding to this enum would mean updating the auto-generator and regenerating all places where this shows up.

Reply:
I don't have a preference here. I would think it should be named `AVX512F` even if it is a separate define/detection/etc.

Reply:
I totally missed the fact that the enum is autogenerated - back to the drawing board... Agree with naming it AVX512F, of course.